snap/integrity/dmverity/veritysetup.go

```diff
@@ -0,0 +1,76 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package dmverity
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/snapcore/snapd/logger"
+	"github.com/snapcore/snapd/osutil"
+)
+
+// Info represents the dm-verity related data that:
+// 1. are not included in the superblock which is generated by default when running
+//    veritysetup.
+// 2. need their authenticity verified prior to loading the integrity data into the
+//    kernel
+//
+// For now, since we are keeping the superblock as it is, this only includes the root hash.
+type Info struct {
+	RootHash string `json:"root-hash"`
+}
+
+func getRootHashFromOutput(output []byte) (string, error) {
+	rootHash := ""
+
+	for _, line := range strings.Split(string(output), "\n") {
+		if strings.HasPrefix(line, "Root hash") {
+			val := strings.SplitN(line, ":", 2)[1]
+			rootHash = strings.TrimSpace(val)
+		}
+	}
+
+	if len(rootHash) == 0 {
+		return "", fmt.Errorf("empty root hash")
+	}
+
+	return rootHash, nil
+}
+
+// Runs veritysetup format and returns an Info struct which include the root hash.
+func Format(dataDevice string, hashDevice string) (*Info, error) {
+	cmd := exec.Command("veritysetup", "format", dataDevice, hashDevice)
+
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return nil, osutil.OutputErr(output, err)
+	}
+
+	logger.Debugf("%s", string(output))
```

Comment:
this probably needs a prefix to clarify what it's the output of

Reply:
Added.